service-front/module/Application/src/Model/Service/Lpa/ContinuationSheets.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Application\Model\Service\Lpa;
+
+use Application\Model\Service\AbstractService;
+use Opg\Lpa\DataModel\Lpa\Formatter as LpaFormatter;
+use Opg\Lpa\DataModel\Lpa\Lpa;
+
+class ContinuationSheets
+{
+    /**
+     * Gathers an array on conditions where continuation sheet(s) would be generated in the PDF.
+     *
+     * @param Lpa $lpa
+     * @return array
+     */
+    public function getContinuationNoteKeys(Lpa $lpa): array
+    {
+        //Array of keys to know which extra notes to show in template for continuation sheets
+        $continuationNoteKeys = array();
+        $extraBlockPeople = null;
+        $paCount = count($lpa->document->primaryAttorneys);
+        $raCount = count($lpa->document->replacementAttorneys);
+        $pnCount = count($lpa->document->peopleToNotify);
+
+        if ($paCount > 4 && $raCount > 2 && $pnCount > 4) {
+            $extraBlockPeople = 'ALL_PEOPLE_OVERFLOW';
+        } elseif ($paCount > 4 && $raCount > 2) {
+            $extraBlockPeople =  'ALL_ATTORNEY_OVERFLOW';
+        } elseif ($paCount > 4 && $pnCount > 4) {
+            $extraBlockPeople =  'PRIMARY_ATTORNEY_AND_NOTIFY_OVERFLOW';
+        } elseif ($raCount > 2 &&  $pnCount > 4) {
+            $extraBlockPeople =  'REPLACEMENT_ATTORNEY_AND_NOTIFY_OVERFLOW';
+        } elseif ($paCount > 4) {
+            $extraBlockPeople =  'PRIMARY_ATTORNEY_OVERFLOW';
+        } elseif ($raCount > 2) {
+            $extraBlockPeople =  'REPLACEMENT_ATTORNEY_OVERFLOW';
+        } elseif ($pnCount > 4) {
+            $extraBlockPeople =  'NOTIFY_OVERFLOW';
+        }
+
+        if ($extraBlockPeople != null) {
+            array_push($continuationNoteKeys, $extraBlockPeople);
+        }
+
+        if ($paCount > 4 || $raCount > 2 || $pnCount > 4) {
+            array_push($continuationNoteKeys, 'ANY_PEOPLE_OVERFLOW');
+        }
+
+        if (
+            isset($lpa->document->primaryAttorneyDecisions->howDetails) ||
+            isset($lpa->document->replacementAttorneyDecisions->howDetails) ||
+            isset($lpa->document->replacementAttorneyDecisions->when)
+        ) {
+            array_push($continuationNoteKeys, 'HAS_ATTORNEY_DECISIONS');
+        }
+
+        if (isset($lpa->document->donor)) {
```

Comment:
Conditions could be combined into one line.

```
if (isset($lpa->document->donor) && !$lpa->document->donor->canSign) {
```

The second part of the condition won't be executed if the first part fails.